x-pack/elastic-agent/docs/inputs/input-logfile.asciidoc

```diff
@@ -0,0 +1,163 @@
+:type: logfile
+
+[id="{beatname_lc}-input-{type}"]
+=== Logfile input
+
+++++
+<titleabbrev>Logfile</titleabbrev>
+++++
+
+Use the `logfile` input to read lines from log files.
+
+To configure this input, specify a list of glob-based <<input-paths,`paths`>>
+that must be crawled to locate and fetch the log lines.
+
+Example configuration:
+
+["source","yaml",subs="attributes"]
+----
+inputs:
+- type: logfile
+  streams:
+    - paths:
+      - /var/log/message
+      - /var/log/*.log
+----
+
+
+You can apply additional
+<<{beatname_lc}-input-{type}-options,configuration settings>> (such as `fields`,
+`include_lines`, `exclude_lines`, `multiline`, and so on) to the lines harvested
+from these files. The options that you specify are applied to all the files
+harvested by this input.
+
+To apply different configuration settings to different files, you need to define
+multiple stream or input sections:
+
+["source","yaml",subs="attributes"]
+----
+inputs:
+  - type: logfile
+    streams:
+     - paths:
+       - /var/log/system.log
+       - /var/log/wifi.log
+     - paths: "/var/log/apache2/*"
+       fields:
+         apache: true
+       fields_under_root: true
+----
+
+<1> Harvests lines from two files:  `system.log` and
+`wifi.log`.
+<2> Harvests lines from every file in the `apache2` directory, and uses the
+`fields` configuration option to add a field called `apache` to the output.
+
+
+IMPORTANT: Make sure a file is not defined more than once across all inputs or stream
+because this can lead to unexpected behaviour.
+
+[[file-identity]]
+==== Reading files on network shares and cloud providers
+
+:WARNING: {beatname_uc} does not support reading from network shares and cloud providers.
+
+However, one of the limitations of these data sources can be mitigated
+if you configure {beatname_uc} adequately.
+
+By default, {beatname_uc} identifies files based on their inodes and
+device IDs. However, on network shares and cloud providers these
+values might change during the lifetime of the file. If this happens
+{beatname_uc} thinks that file is new and resends the whole content
```

Comment:
I think this was my fault, but usually we don't anthropomorphize software components. :-) Maybe say: "If this happens, Elastic Agent processes the file as new and resends...." I don't have a strong feeling, just pointing out the general rule.